DOCUMENTATION.md

```diff
@@ -43,6 +45,153 @@ The overall territory is laid out here in this [C4](https://c4model.com) System
+- DynamoDb: While DynamoDb streams establishes and preserves the order of each change, that's subject to the 24h retention period limit. The DynamoStore.Indexer and reader all-but preserve that ordering (Note that the indexer does not currently ensure to preserve order _within_ any DDB Streams batch being indexed).
+- MessageDb: Appends establish a permanent position within the events for a given category (writes for a given category are serialized in order to guarantee this; there are explicitly no guarantees across categories). The Propulsion reader logic pulls categories independently (which should not matter due to the lack of correlation within the store).
+- MemoryStore: `MemoryStore`, and `Propulsion.MemoryStore` explicitly guarantees that notifications of write for a given stream are processed (and propagated into the Scheduler) in strict order of those writes (all Propulsion Sources are expected to guarantee that Stream level ordering). 
+
+While the above factors are significant in how the Propulsion Sources are implemented, **the critical thing to understand is that `Propulsion.Streams` makes no attempt to preserve any ordering beyond the individual stream**.
+
+The primary reason for this is that it would imply that invocation of handlers cannot be concurrent. That would be a major impediment to throughput when rate limiting and/or other bottlenecks impose non-uniform effects on handler latency (you'd ideally continue processing on streams that are not currently impacted by rate limiting or latency spikes, working ahead on the basis that the impeded stream's issues will eventually abate).
+
+The other significant opportunity one leaves behind if you don't admit concurrency into your projections is that you can never split or load balance the processing across multiple consumers (Propulsion does not presently implement any explicit sharding support, although when using `Propulsion.CosmosStore`, the underlying Change Feed Processor in the Microsoft SDK implements automated balancing of physical partition leases across all competing instances).
+
+While doing the simplest thing possible to realise any aspect of a system's operation is absolutely the correct starting point for any design, it's also important to consider whether such solutions might impose restrictions that may later be difficult or costly to unravel:
+- re-traversing of all events to build a new version of a read model becomes less viable if the time to do that is hardwired to be limited by the serial processing of the items
+- while grouping multiple operations into a single SQL batch in order to reduce the latency (although not the locking side effects) can be effective, it should be noted that such a scheme does not generalise well to the e.g. third party API calls, or writes to document stores etc that do not present such facilities.
```

Comment:
I'd say this is more about throughput than latency
```suggestion
- while grouping multiple operations into a single SQL batch in order to increase throughput can be effective, it should be noted that such a scheme does not generalise well to the e.g. third party API calls, or writes to document stores etc that do not present such facilities.
```

Reply:
Yes, silly me! I added back on a reference about batching not being free - i.e. increased chances of lock escalation and deadlocks